### ISSUE TITLE
[Feature request] Trbr macros repository

### ISSUE THREAD

**Issue by mensfeld** [trailblazer/trailblazer#167]
Hey, I've noticed that you encourage people to use macros (http://trailblazer.to/gems/operation/2.0/api.html#step-macros) for their stateless steps that can be reused. Great idea (been using that with "containers" namespace before ;) ) and what I would recommend is to extract this part of trbr into a separate gem as now it lies under trbr::operation::model trbr::operation::whatever - maybe it should be moved (to enforce unification) to something like Trailblazer::Macros::Model, Trailblazer::Macros::Contract, etc with some sort of simple api or just recommended style as an example to the whole community. Now I personally  wrap everything like that (as a separate shared gem across multiple system):

```ruby
# frozen_string_literal: true
module Repositories
  # Creating repository operation
  class Create < ApplicationOperation
    contract Contracts::Create

    step Macros::Model::Build(Repository)
    step Macros::Contract::Build()
    step Macros::Contract::Validate(key: :repository)
    step Macros::Contract::Persist()

    failure Macros::Error::Raise(Errors::OperationFailure)
  end
end
```

and it works great

**Comment by apotonick** [comment]
The original idea was that everything under `Trailblazer::Operation` is a macro (with a handful of reserved constants, like `Railway`), hence the absence of a qualifying constant.

The other idea was that the `trailblazer` gem gives us all the nice standard macros. I'm actually thinking about moving `Nested`, `Wrap`, etc. to `trailblazer-operation`.

**Comment by mensfeld** [comment]
Yeah now it seems (as I've stated earlier) a bit inconsistent and I still stand by moving it to a ```Trailblazer::Macros``` (or something with Macros namespace) to keep things consistent. 

You call it "Macro API", you name it Macro, you use the ```Macro``` module in the example and then you put your own macros in the ```Trailblazer::Operation``` namespace without indicating that those are macros as well (primary). For me that doesn't make much sense and creates a confusing element to which users will also have to adapt (and understand why it is like that).

If not a separate gem, then maybe just a Macro namespace - still would be consistent.

**Comment by apotonick** [comment]
Again, the goal is that at some point everything is just a macro, and then the `Macro` namespace will be super annoying as you need to specify it literally everywhere, redundantly.

**Comment by mensfeld** [comment]
@apotonick  I agree that this is annoying but on the other hand it:

- is consistent with the 2.0 macro api naming convention
- would be redundant but would instruct people to follow the same principle in their code

I don't mind keeping that in the operation namespace - I just  find that inconsistent. Having one style would allow people to create their gems in the future for integration like ```trailblazer-ransack``` with ```Macros::Model::Search```, ``Macros::Model::Paginate``` or whatever they need to do with models inside their pipie flow.
